modules/data.land/R/pool_ic_list2netcdf.R

```diff
@@ -0,0 +1,98 @@
+##' @name pool_ic_list2netcdf
+##' @title pool_ic_list2netcdf
+##' @export
+##'
+##' @param input list with two elements: list of netcdf dimensions (dims, with named values) and list of variables (vals, with named values)
+##' @param outdir directory to write netcdf file
+##' @param siteid site id
+##' @author Anne Thomas
+
+pool_ic_list2netcdf <- function(input, outdir,siteid){
+  if(is.null(input$dims) || length(input$dims) == 0){
+    PEcAn.utils::logger.severe("Please provide non-empty 'dims' list in input")
+  }
+  if(is.null(input$vals) || length(input$vals) == 0){
+    PEcAn.utils::logger.severe("Please provide 'vals' list in input with variable names assigned to values")
+  }
+  
+  standard_vars <- read.csv(system.file("data/standard_vars.csv",package="PEcAn.utils"),stringsAsFactors = FALSE)
+  
+  ###function to lapply to all dimensions in input list and return a list of ncdims
+  to_ncdim <- function(dimname,standard_vars,input){
+    dim <- standard_vars[which(standard_vars$Variable.Name == dimname),]
+    #check dim exists
+    if(nrow(dim) == 0){
+      PEcAn.utils::logger.severe(paste("Dimension",dimname,"not in standard_vars"))
+    }
+    if(dim$Category != "Dimension"){
+      PEcAn.utils::logger.severe(paste(dimname,"not a dimension or is deprecated"))
+    }
+    
+    vals <- input$dims[[which(names(input$dims) == dimname)]] #makes function non-modular but works assuming it's only used in pool_ic_list2netcdf
+    if(is.null(vals) || length(vals) == 0){
+      PEcAn.utils::logger.severe(paste("Missing vals for dim",dimname,",please check input")) #don't know if this could happen without explicit NULL in input
+    }
+    
+    units <- as.character(dim$Units) #if the units are a factor the function fails
```

Comment:
Isn't this already guaranteed by the `stringsAsFactors = FALSE` when reading standard_vars?